ReadASCIIApp/src/ReadASCII.cpp

```diff
@@ -326,24 +312,27 @@ asynStatus ReadASCII::readFloat64Array(asynUser *pasynUser, epicsFloat64 *value,
     int function = pasynUser->reason;
     asynStatus status = asynSuccess;
     const char *functionName = "readFloat64Array";
-
-    ncopy = ( (nElements < rowNum) ? nElements : rowNum );
-    if (function == P_SPArr) {
-        memcpy(value, pSP_, ncopy*sizeof(epicsFloat64));
-    }
-    else if (function == P_PArr) {
-        memcpy(value, pP_, ncopy*sizeof(epicsFloat64));
-    }
-    else if (function == P_IArr) {
-        memcpy(value, pI_, ncopy*sizeof(epicsFloat64));
-    }
-    else if (function == P_DArr) {
-        memcpy(value, pD_, ncopy*sizeof(epicsFloat64));
-    }
-    else if (function == P_MaxHeatArr) {
-        memcpy(value, pMaxHeat_, ncopy*sizeof(epicsFloat64));
+    std::unordered_map<std::string, int>::iterator it;
+    for (it = pv_index_arrays.begin(); it != pv_index_arrays.end(); it++)
+    {
+        if (pv_data_column_reference.find(it->first) != pv_data_column_reference.end())
+        {
+            try
+            {
+                if (function == it->second) {
+                    memcpy(value, pv_data_column_reference[it->first]->data(), ncopy * sizeof(epicsFloat64));
+                }
+            }
+            catch (std::out_of_range)
+            {
+                std::cerr << FILE_FORMAT_INCORRECT << std::endl;
+            }
+        }
+        else
+        {
+            std::cerr << "ReadASCII: WARNING - parameter name: " << it->first << " doesn't have associated table column" << std::endl;
+        }
     }
-    *nIn = ncopy;
```

Comment:
You need to set `*nIn` - this tells epics how many values you returned. On success `*nIn = ncopy`, on error you should set `*nIn = 0`